src/cmd/run.rs

```diff
@@ -104,9 +114,50 @@ pub async fn main<D: Database>(ctx: &AppContext, config: Config, db: D) -> Resul
         db,
     };
 
+    let preflight_checks = preflight_checks(&ctx);
     let healthcheck_worker = healthcheck(&ctx.db);
     let integration_worker = integrations(&ctx);
-    try_join!(healthcheck_worker, integration_worker).discard_ok()
+    try_join!(preflight_checks, healthcheck_worker, integration_worker).discard_ok()
+}
+
+/// Checks and catches network misconfigurations before Broker attempts its operations
+async fn preflight_checks<D: Database>(ctx: &CmdContext<D>) -> Result<(), Error> {
+    let validate_integration_connections = integration_connections(ctx.config.integrations());
+    let validate_fossa_connection = fossa_connection(&ctx.config);
+    try_join!(validate_integration_connections, validate_fossa_connection).discard_ok()
+}
+
+#[tracing::instrument(skip_all)]
+/// Validate that Broker can connect to at least one integration
+async fn integration_connections(integrations: &Integrations) -> Result<(), Error> {
+    if integrations.as_ref().is_empty() {
+        return Ok(());
+    }
+
+    for integration in integrations.iter() {
+        let Protocol::Git(transport) = integration.protocol();
+        if repository::ls_remote(transport).await.is_ok() {
+            return Ok(());
+        }
+    }
+
+    report!(Error::IntegrationConnection)
+        .wrap_err()
+        .help("run broker fix for detailed explanation on failing integration connections")
+        .describe("integration connections")
+}
+
+#[tracing::instrument(skip_all)]
+/// Validate that Broker can connect to FOSSA
+async fn fossa_connection(config: &Config) -> Result<(), Error> {
+    match fossa::OrgConfig::lookup(config.fossa_api()).await {
+        Ok(_) => Ok(()),
+        Err(err) => err
+            .change_context(Error::FossaConnection)
+            .wrap_err()
+            .help("ensure that your fossa key is configured correctly")
```

Comment:
This error message is good for the case that the key is not configured correctly, but what about if the error is something else? At this point in the code there's no way to say for certain that the issue is a fossa api key error.

In general, you should try to give error messages using only information we have available to you in context. You don't have enough information to say for certain that the error was an issue with the fossa api key. [Further down the call-stack](https://github.com/fossas/broker/blob/ANE-904-Preflight-Check/src/api/fossa.rs#L421) messages about a bad API key will already be written because that's where that error is constructed. So in this case, you don't have to do anything for advice like this to already be applied. 

I think you can reduce this function down to just the call to `lookup` (or just call it inline in `preflight_checks`) and then apply `change_context` with `Error::FossaConnection`. Additionally, it might be good to create a new `Error::PreflightCheck` and apply that context on top of the `FossaConnection` or `IntegrationConnection`. That way, the error will render as there being a preflight check, and then inside that the context will be the type of issue.

Reply:
Would it be helpful for the `.help` if I said something like `run broker fix for a detailed explanation on the failing fossa connection` similar to what I have for `check_integration_connections`? Broker fix is more detailed and provides explanations/solutions to the specific problems

Reply:
Broker fix isn't currently recommended in the output. I've tested some scenarios on failing fossa connection and Broker fix provides actionable advice for them. 

Tested on :
- Bad fossa_key
- Malformed fossa_endpoint url